### DISCUSSION TITLE
mariadbconnector doesn't pass along query parameters in the url as arguments to connect method as mysqldb does

### DISCUSSION THREAD

**Discussion by tapetersen** [sqlalchemy/sqlalchemy#11868] [Usage Questions] [answered]
While trying to ensure our codebase works with both connectors the first issue was that appending, in our case, `?unix_socket=/path/to/socket` wasn't having any effect while using mariadbconnector while mysqldb passes it as args to the connect method.

Both drivers are very similar and supports the same argument in almost all cases so I assume it would work fine just doing the same call in both drivers (or if that is not a desired usage deprecate passing it as url_params and document that it's prefered to pass it as explicit options to create_engine).

We can work around it either way but it feels like an unnecesary inconsistency.

I havn't checked the behavior of other dialects in this regard but the difference between these to is simply the call to `opts.update(url.query)` at 
https://github.com/sqlalchemy/sqlalchemy/blob/29c2c15902094965f9dca66efae33305f09c3878/lib/sqlalchemy/dialects/mysql/mysqldb.py#L208

which has no corresponding call in mariadbconnector.
Just adding it here https://github.com/sqlalchemy/sqlalchemy/blob/29c2c15902094965f9dca66efae33305f09c3878/lib/sqlalchemy/dialects/mysql/mariadbconnector.py#L169

works reasonably in my preliminary testing.

**Comment by zzzeek** [comment, marked as answer]
this is likely a bug since we are coercing for int and bool params, but coercing is not needed for string parameters like unix socket so there should have been an update() call ahead of those coerce calls as you observe for mariadb.   

we can accept a PR to expedite if you have the motivation

**Reply by tapetersen** [reply]
Sure I'll take a shot at that. 
Is there some guidelines of extra tests or documentation or just the oneliner that should be the real fix in this case?

**Reply by zzzeek** [reply]
sure, we're going to be in this file: https://github.com/sqlalchemy/sqlalchemy/blob/main/test/dialect/mysql/test_dialect.py

and examples of tests that are already there for this kind of thing can be seen such as for mysqlclient:

https://github.com/sqlalchemy/sqlalchemy/blob/29c2c15902094965f9dca66efae33305f09c3878/test/dialect/mysql/test_dialect.py#L254-L276

we could likely add "unix_socket" to the list of combinations for that test and even repurpose that single test to also do mariadbconnector (using an additional `@testing.combinations` on top), or add a test right below it that is more specific to mariadbconnector

an example of a test that's for mariadbconnector directly is:

https://github.com/sqlalchemy/sqlalchemy/blob/29c2c15902094965f9dca66efae33305f09c3878/test/dialect/mysql/test_dialect.py#L358-L364


you can run that test suite directly by installing mariadbconnector; if you have a mariadb database running on localhost with username/password "scott / tiger" database "test" you can run:

    pytest test/dialect/mysql/test_dialect.py  --db mariadb_connector

or a specific URL

     pytest test/dialect/mysql/test_dialect.py  --dburi mariadb+mariadbconnector://scott:tiger@mariadb111/test

**Reply by tapetersen** [reply]
Thanks and I found those. I have added an issue in #11870 and a PR in #11869 so it's probably better to continue discussion there.

**Comment by tapetersen** [comment]
Solved in the corresponding issue.
